components/server/src/analytics.ts

```diff
@@ -38,10 +46,23 @@ export async function trackSignup(user: User, request: Request, analytics: IAnal
     });
 }
 
-function fullIdentify(user: User, request: Request, analytics: IAnalyticsWriter) {
-    //makes a full identify call for authenticated users
+async function fullIdentify(
+    user: User,
+    request: Request,
+    analytics: IAnalyticsWriter,
+    subscriptionService?: SubscriptionService,
+) {
+    // makes a full identify call for authenticated users
     const coords = request.get("x-glb-client-city-lat-long")?.split(", ");
     const ip = request.get("x-forwarded-for")?.split(",")[0];
+    const subscriptions = subscriptionService
+        ? (
+              (await subscriptionService.getNotYetCancelledSubscriptions(
+                  user,
+                  new Date().toISOString(),
+              )) as Subscription[]
+          ).map((sub) => sub.planId)
+        : undefined;
```

Comment:
I think here we either shift the `undefined` logic into the `getNotYetCancelledSubscriptions` method, such that it never throws and always returns either `undefined` or `Subscription[]`. That way, you can remove the logic here.

Alternatively, I'd find this slightly easier to visually grep, but that's just a suggestions and doesn't block this PR.
```typescript
var subscriptionIDs: string[] | undefined = undefined;
const subscriptions = await subscriptionService.getNotYetCancelledSubscriptions(user, new Date().toISOString());
if (subscriptions) {
	subscriptionIDs = subscriptions.map(...)
}
```

Reply:
thanks, your suggestion makes it much easier to follow! i used it, but had to account for cases in which planId is undefined - hope this does not destroy all of the gained clarity😃